rfc-drafts/stream.md

```diff
@@ -233,19 +247,62 @@ Designing such a migration feature is out of scope for this RFC.
 
 ## Async iteration syntax
 
-We may wish to introduce some dedicated syntax, analogous to `for` 
+Currently, if someone wishes to iterate over a `Stream` as defined in the `futures` crate,
+they are not able to use  `for` loops, they must use `while let` and `next/try_next` instead.
+
+We may wish to extend the `for` loop so that it works over streams as well. 
+
+```rust
+#[async]
+for elem in stream { ... }
+```
+
+Designing this extension is out of scope for this RFC.
+
+## "Attached" streams
+
+There has been much discussion around attached/detached streams.
```

Comment:
> every "Detached" steam can become an attached one, and _some_ detached streams can become attached ones, but that is harder to specify

I'm not quite sure I follow – I think you meant that some attached streams can become detached streams? In that case it seems clear that you would define it as `impl Stream` from the start, since that's more general?